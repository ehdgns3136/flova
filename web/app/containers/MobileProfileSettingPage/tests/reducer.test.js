
import { fromJS } from 'immutable';
import mobileProfileSettingPageReducer from '../reducer';
import {
  getProfileRequest,
  getProfileSuccess,
  getProfileFailure,
} from '../actions';

describe('mobileProfileSettingPageReducer', () => {
  it('returns the initial state', () => {
    expect(mobileProfileSettingPageReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle getProfileRequest action correctly', () => {
    const state1 = fromJS({
      isMobileProfileSettingPageError: false,
      isMobileProfileSettingPageLoading: false,
      profile: null,
    });

    const stateResult1 = fromJS({
      isMobileProfileSettingPageError: false,
      isMobileProfileSettingPageLoading: true,
      profile: null,
    });

    expect(mobileProfileSettingPageReducer(state1, getProfileRequest(1))).toEqual(stateResult1);
  });

  it('should handle getProfileSuccess action correctly', () => {
    const profile = {
      id: 1,
      name: 'HongGilDong',
    };

    const state1 = fromJS({
      isMobileProfileSettingPageError: false,
      isMobileProfileSettingPageLoading: true,
      profile: null,
    });

    const stateResult1 = fromJS({
      isMobileProfileSettingPageError: false,
      isMobileProfileSettingPageLoading: false,
      profile,
    });

    expect(mobileProfileSettingPageReducer(state1, getProfileSuccess(profile))).toEqual(stateResult1);
  });

  it('should handle getProfileFailure action correctly', () => {
    const state1 = fromJS({
      isMobileProfileSettingPageError: false,
      isMobileProfileSettingPageLoading: true,
      profile: null,
    });

    const stateResult1 = fromJS({
      isMobileProfileSettingPageError: true,
      isMobileProfileSettingPageLoading: false,
      profile: null,
    });

    expect(mobileProfileSettingPageReducer(state1, getProfileFailure())).toEqual(stateResult1);
  });
});
