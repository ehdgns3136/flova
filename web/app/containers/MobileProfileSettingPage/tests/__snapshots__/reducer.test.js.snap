exports[`mobileProfileSettingPageReducer returns the initial state 1`] = `
Object {
  "isMobileProfileSettingPageError": false,
  "isMobileProfileSettingPageLoading": false,
  "profile": null,
}
`;
